### ISSUE TITLE
multiline_grok not working as expected with td-agent v3.1.1 for windows 2012 R2

### ISSUE THREAD

**Issue by bcharboneauiherb** [fluent/fluent-plugin-grok-parser#42]
This may or may not be user error but I would appreciate some help.

Message is not being separated into multiple key value pairs as defined by the grok pattern. but the lines are being concatenated correctly.  Multiline aspect works grok matching does not appear to be functioning.

Config
```
<source>
  @type tail
  path C:/Logs/*
  pos_file logs.log.pos
  <parse>
      @type multiline_grok
      grok_pattern \[(?<sourceTimestamp>%{DATE} %{TIME} (AM|PM))\]\t(?<tempMessage>(.|\r|\n)*)
      multiline_start_regexp /\[[0-9]{1,2}\/[0-9]{1,2}\/[0-9]{4}\s[0-9]{1,2}\:[0-9]{2}\:[0-9]{2}\s(?:A|P)M\]/
  </parse>
    tag logs
</source>
<match **>
    @type copy
    # Output to Console
    <store>
      @type stdout
    </store>
  </match>
```
Sample Log Entry
```
[2/16/2018 10:19:34 AM] Message Line 1
Message Line 2
Message Line 3
```
Received Output
```
2018-02-16 11:00:50.115236000 -0800 logs: {"message":"[2/16/2018 10:19:34 AM]\tMessage Line 1\r\nMessage Line 2\r\nMessage Line 3\r\n"}
```
As you can see we only get the entire log concatenated into "**message**" instead of it being concatenated and split into "**sourceTimestamp**" and "**tempMessage**"

**Comment by okkez** [comment]
You can use `DATE_US` instead of `DATE` like following:

```
<source>
  @type tail
  path 42.txt
  pos_file logs.log.pos
  read_from_head true
  <parse>
    @type multiline_grok
    grok_pattern \[(?<sourceTimestamp>%{DATE_US} %{TIME} (AM|PM))\]\t(?<tempMessage>(.|\r|\n)*)
    multiline_start_regexp /\[[0-9]{1,2}\/[0-9]{1,2}\/[0-9]{4}\s[0-9]{1,2}\:[0-9]{2}\:[0-9]{2}\s(?:A|P)M\]/
  </parse>
  tag logs
</source>
```

This is a workaround for you.

This issue is caused by https://github.com/logstash-plugins/logstash-patterns-core/blob/master/patterns/grok-patterns#L72
@bcharboneauiherb  Could you create new issue or PR on https://github.com/logstash-plugins/logstash-patterns-core ?

**Comment by okkez** [comment]
Also you can fix pattern like following:

Before:

```
DATE %{DATE_US}|%{DATE_EU}
```

After:

```
DATE (?:%{DATE_US}|%{DATE_EU})
```

**Comment by bcharboneauiherb** [comment]
Thanks, I will create the new issue for logstash-patterns-core. so far using DATE_US is working in parsing the data.

**Comment by okkez** [comment]
Thanks!!

https://github.com/logstash-plugins/logstash-patterns-core/issues/221

**Comment by bcharboneauiherb** [comment]
@okkez can you help with communicating the issue on the logstash-patterns-core git? they seem to be dismissing it as your issue.

**Comment by okkez** [comment]
I've already fixed issue and released new version v2.1.6.

Thank you very match for creating issue there!!
